spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SkipGramFeatureGenerator.java

```diff
@@ -0,0 +1,173 @@
+    StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
+    Schema inputSchema = stageConfigurer.getInputSchema();
+    Preconditions.checkArgument(inputSchema != null, "Input Schema must be a known constant.");
+    pipelineConfigurer.getStageConfigurer().setOutputSchema(config.getOutputSchema(inputSchema));
+    SparkUtils.validateFeatureGeneratorConfig(inputSchema, config.outputColumnMapping);
+  }
+
+  @Override
+  public void initialize(SparkExecutionPluginContext context) throws Exception {
+    super.initialize(context);
+    FileSet fileSet = context.getDataset(config.fileSetName);
+    Location modelLocation = fileSet.getBaseLocation().append(config.path);
+    if (!modelLocation.exists()) {
+      throw new IllegalArgumentException(String.format("Failed to find model. Location does not exist: %s.",
+                                                       modelLocation));
+    }
+    // load the model from a file in the model fileset
+    loadedModel = Word2VecModel.load(context.getSparkContext().sc(), modelLocation.toURI().getPath());
+  }
+
+  @Override
+  public JavaRDD<StructuredRecord> transform(SparkExecutionPluginContext context,
+                                             JavaRDD<StructuredRecord> input) throws Exception {
+    if (input.isEmpty()) {
+      return context.getSparkContext().emptyRDD();
+    }
+    outputSchema = outputSchema != null ? outputSchema : config.getOutputSchema(input.first().getSchema());
+    final Map<String, String> mapping = config.getFeatureListMapping();
+
+    final int vectorSize = loadedModel.wordVectors().length / loadedModel.wordIndex().size();
+    return input.map(new Function<StructuredRecord, StructuredRecord>() {
+      @Override
+      public StructuredRecord call(StructuredRecord input) throws Exception {
+        StructuredRecord.Builder builder = StructuredRecord.builder(outputSchema);
+        for (Schema.Field field : input.getSchema().getFields()) {
+          String fieldName = field.getName();
+          builder.set(fieldName, input.get(fieldName));
+        }
+        for (Map.Entry<String, String> mapEntry : mapping.entrySet()) {
+          String inputField = mapEntry.getKey();
+          String outputField = mapEntry.getValue();
+          List<String> text = SparkUtils.getInputFieldValue(input, inputField, config.pattern);
+          int numWords = 0;
+          double[] vectorValues = new double[vectorSize];
+          for (String word : text) {
+            double[] wordVector = loadedModel.transform(word).toArray();
+            for (int i = 0; i < vectorSize; i++) {
+              vectorValues[i] += wordVector[i];
+            }
+            numWords++;
+          }
+          for (int i = 0; i < vectorSize; i++) {
+            vectorValues[i] /= (double) numWords;
+          }
+          builder.set(outputField, new ArrayList<>(Arrays.asList(ArrayUtils.toObject(vectorValues))));
```

Comment:
Directly setting the field to double[] throws ``java.io.IOException: Incomplete document``
Therefore, converting double[] to Objects[].
Please let me know if this is okay or if I'm missing anything.